### ISSUE TITLE
 Convergence detected: absolute parameter change was below tolerance issue

### ISSUE THREAD

**Issue by tauseef-khan** [facebookincubator/prophet#275]
Here is my dataframe:

```
           ds        y
1    15/07/2014   0.99
2    30/07/2014   1.34
3    19/09/2014   1.16
4    12/05/2015   1.23
5    03/11/2015   1.23
6    08/02/2016   1.24
9    17/06/2016   0.98
10   26/06/2016   1.02
11   10/07/2016   1.24
12   23/09/2016   1.31
14   01/11/2016   1.43
```

When fitting the model:
_> m <- prophet(df4)_

I get the following output:

_Initial log joint probability = -2.16647
Optimization terminated normally: 
  Convergence detected: absolute parameter change was below tolerance
Warning message:
In set_auto_seasonalities(m) :
  Disabling weekly seasonality. Run prophet with `weekly.seasonality=TRUE` to override this._

Then when I create dataframe to store the predictions, I get this:

```
> future <- make_future_dataframe(m, periods = 1)
> tail(future)
```
```
          ds
7    0017-06-20
8    0019-09-20
9    0023-09-20
10   0026-06-20
11   0030-07-20
12   0030-07-21
```

So it changes the dates to some other ones.

Then when predicting I get this output:

```
> forecast <- predict(m, future)
> tail(forecast[c('ds', 'yhat', 'yhat_lower', 'yhat_upper')])
```
```
           ds      yhat       yhat_lower   yhat_upper
7  0017-06-20   0.9973871    0.8821676     1.112004
8  0019-09-20   1.2357150    1.1247759     1.345855
9  0023-09-20   1.2408199    1.1305598     1.355770
10 0026-06-20   1.0098083    0.8984269     1.122859
11 0030-07-20   1.2105866    1.1067807     1.326290
12 0030-07-21   1.2130727    1.0964266     1.324312
```

The _yhat_ values are different to the original ones.

Can anyone help me with the "Convergence detected issue" and the changing date and yhat values from the original values?

**Comment by bletham** [comment]
The "Convergence detected" is a message from the underlying Stan optimizer and is normally - it is actually indicating that the model was successfully fit. 

It does look like the dates were not parsed correctly. Prophet currently uses `zoo::as.Date(df$ds)` to parse the dates - if you do this on your dataframe you will see that they come out a mess.

This issue is actually already fixed in the v0.2 branch which will be released soon (it raises an error in this case), but in the meantime you just need to convert the `ds` column to be a correct date type prior to passing it into Prophet, by specifying the correct format: http://www.statmethods.net/input/dates.html

**Comment by tauseef-khan** [comment]
Thank you for your help, it's been very useful!
